collect_app/src/main/java/org/odk/collect/android/preferences/AggregatePreferencesActivity.java

```diff
@@ -25,26 +28,44 @@
 import android.preference.PreferenceManager;
 import android.text.InputFilter;
 import android.text.Spanned;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.ListPopupWindow;
 import android.widget.Toast;
 
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
 import org.odk.collect.android.R;
 import org.odk.collect.android.application.Collect;
 import org.odk.collect.android.utilities.AuthDialogUtility;
 import org.odk.collect.android.utilities.UrlUtils;
 import org.odk.collect.android.utilities.WebUtils;
 
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Handles aggregate specific preferences.
  *
  * @author Carl Hartung (chartung@nafundi.com)
  */
 public class AggregatePreferencesActivity extends PreferenceActivity {
 
+    private static final String KNOWN_URL_LIST = "knownUrlList";
+
     protected EditTextPreference mServerUrlPreference;
     protected EditTextPreference mUsernamePreference;
     protected EditTextPreference mPasswordPreference;
     protected boolean mCredentialsHaveChanged = false;
 
+    private ListPopupWindow mListPopupWindow;
+    private List<String> mUrlList;
```

Comment:
And actually, same question for mAdapter. Would be better to have local variables wherever possible.